Amethyst/Layout/ReflowOperation.swift

```diff
@@ -62,6 +62,177 @@ open class ReflowOperation: Operation {
         }
     }
 
+    open func isDegenerateWindow(window: SIWindow) -> Bool {
+        let window_frame = window.frame()
+        if window_frame.origin.x.isInfinite {
+            return true
+        }
+        if window_frame.origin.x.isNaN {
+            return true
+        }
+        if window_frame.origin.y.isInfinite {
+            return true
+        }
+        if window_frame.origin.y.isNaN {
+            return true
+        }
+        if window_frame.width <= 0.0 {
+            return true
+        }
+        if window_frame.width.isInfinite {
+            return true
+        }
+        if window_frame.width.isNaN {
+            return true
+        }
+        if window_frame.height <= 0.0 {
+            return true
+        }
+        if window_frame.height.isInfinite {
+            return true
+        }
+        if window_frame.height.isNaN {
+            return true
+        }
+        return false
+    }
+
+    private func categorizeWindows(candidateFrames: [CGRect]) -> ([SIWindow], [SIWindow]) {
+
+        var displaced: [SIWindow] = []
+        var aligned: [SIWindow] = []
+        for window in self.windows {
+            //Sometimes the windows become degenerate for some reason, ignore them
+            if !isDegenerateWindow(window:window) {
+                var min_distance = FLT_MAX
+                let window_frame = window.frame()
+                let windows_center_x = window_frame.origin.x + (window_frame.width/2.0)
+                let windows_center_y = window_frame.origin.y + (window_frame.height/2.0)
+                for frame in candidateFrames {
+                    let frame_center_x = frame.origin.x + (frame.width/2.0)
+                    let frame_center_y = frame.origin.y + (frame.height/2.0)
+                    let distance_x = (frame_center_x - windows_center_x)
+                    let distance_y = (frame_center_y - windows_center_y)
+                    let distance = sqrt(distance_x*distance_x + distance_y*distance_y)
+                    if Float(distance) < min_distance {
+                        min_distance = Float(distance)
+                    }
+                }
+                if min_distance < 1.0 {
```

Comment:
Some windows don't align perfectly. Terminal windows, for example, are quantized to the size of the cursor. Doesn't this mean such windows will be incorrectly categorized as displaced?